src/Microsoft.TemplateEngine.Edge/Settings/TemplatePackageManager.cs

```diff
@@ -235,7 +237,20 @@ private async Task<TemplateCache> UpdateTemplateCacheAsync(bool needsRebuild, Ca
             Task<IReadOnlyList<ITemplatePackage>> getTemplatePackagesTask = GetTemplatePackagesAsync(needsRebuild, cancellationToken);
             if (!(_userTemplateCache is TemplateCache cache))
             {
-                cache = new TemplateCache(JObject.Parse(_paths.ReadAllText(_paths.TemplateCacheFile, "{}")), _logger);
+                try
```

Comment:
should we move the code below to `ReadAllText` (or ReadJson) to benefit from it in other places it is used? All use cases of `ReadAllText` is for reading json.

Reply:
Created new JExtension extension method and call from everywhere...